packages/namekit-react/src/components/TruncatedText.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * This TruncatedText component is a duplicate of @namehash/nameguard-react
+ * TruncatedText component. This duplicate will, in the near future, be the
+ * only TruncatedText component in the NameKit libraries. The duplicate is
+ * necessary for the moment so that it is included in main branch while
+ * @namehash/nameguard-react still has its TruncatedText: why is that?
+ *
+ * So any imports of @namehash/nameguard-react TruncatedText component
+ * do not break once this change is merged into main! Once this duplicate
+ * is in main, NameHash Labs will remove the TruncatedText component from
+ * @namehash/nameguard-react and update the imports to use this component
+ * instead, the @namehash/namekit-react TruncatedText.
+ *
+ * A ShortCut story was created to track this migration:
+ * https://app.shortcut.com/ps-web3/story/25261/next-steps-on-displayedname-and-truncatedtext
+ */
+
+import React, { useEffect, useState, useRef } from "react";
+import { Tooltip } from "./Tooltip";
+import cc from "classcat";
+
+interface TruncatedTextProps {
+  text: string;
+  maxTooltipWidth?: number;
+  maxDisplayWidth?: number;
+  textStylingClasses?: string;
+  tooltipTextStylingClasses?: string;
+  displayTooltipWhenTextOverflows?: boolean;
+}
+
+const DEFAULT_MAX_DISPLAY_WIDTH = 300;
+const DEFAULT_MAX_TOOLTIP_WIDTH = 400;
+
+export const TruncatedText = ({
+  text,
+  textStylingClasses = "",
+  tooltipTextStylingClasses = "",
+  displayTooltipWhenTextOverflows = true,
+  maxTooltipWidth = DEFAULT_MAX_TOOLTIP_WIDTH,
+  maxDisplayWidth = DEFAULT_MAX_DISPLAY_WIDTH,
+}: TruncatedTextProps) => {
+  const invisibleTextWidthTester = useRef<null | HTMLParagraphElement>(null);
+
+  /* 
+    Below state is only true if the text displayed 
+    is longer than text maxDisplayWidth
+  */
+  const [textOverflows, setTextOverflows] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (
+      displayTooltipWhenTextOverflows &&
+      invisibleTextWidthTester &&
+      invisibleTextWidthTester.current
+    ) {
+      const textOverflows =
+        Math.ceil(
+          invisibleTextWidthTester.current.getBoundingClientRect().width,
+        ) > maxDisplayWidth;
+
+      setTextOverflows(textOverflows);
+    }
+  }, [invisibleTextWidthTester]);
+
+  const getTextElm = (
+    classes: string,
+    maxWidth = maxDisplayWidth,
+  ): JSX.Element => {
+    return (
+      <p style={{ maxWidth: maxWidth }} className={classes}>
+        {text}
+      </p>
+    );
+  };
+
+  const textDefaultClasses = "nk-text-black nk-text-sm nk-truncate";
+
```

Comment:
I would add something like this for managing the tooltip:

```ts
const tooltipContent = renderTooltip ? renderTooltip(text) : renderText(
    cc(["nk-text-white nk-text-sm nk-leading-5 nk-break-all", tooltipTextStylingClasses]),
    maxTooltipWidth
  );
```

Reply:
@notrab Thanks for suggestion :+1:

@FrancoAguzzi What do you think? Please share your suggested code updates with us.

Reply:
I liked the idea, @notrab !

@lightwalker-eth , I have updated the code with similar abstraction 🔑

take a look and let me know what you think